homeassistant/components/scrape/config_flow.py

```diff
@@ -193,7 +193,15 @@ async def validate_sensor_edit(
     # Standard behavior is to merge the result with the options.
     # In this case, we want to add a sub-item so we update the options directly.
     idx: int = handler.flow_state["_idx"]
-    handler.options[SENSOR_DOMAIN][idx].update(user_input)
+    sensor_options: dict[str, Any] = handler.options[SENSOR_DOMAIN][idx]
+    sensor_options.update(user_input)
+
+    # These keys are optional, and absent from user_input when unset in UI
```

Comment:
I think it's a bug in frontend if selecting the empty string does not work, we should fix that instead of documenting it here.
Having said that, I also don't think the solution where the select selector value = select selector label = "" is good.
Make the value "none", and provide a translation "No device class" instead.

Reply:
> I think it's a bug in frontend if selecting the empty string does not work, we should fix that instead of documenting it here. I did do https://github.com/home-assistant/frontend/pull/14899 but seems to impact other things so haven't gone anywhere so far.

Reply:
We could use `SelectOptionDict` instead of translations to provide proper labels to not have to deal with changes in `SensorDeviceClass` or `SensorStateClass`

Reply:
If we use a specific "real" value, then we would still need to `pop` them from the options:
```python
    for key in (CONF_DEVICE_CLASS, CONF_STATE_CLASS, CONF_UNIT_OF_MEASUREMENT):
        if user_input[key] == "none":
            sensor_options.pop(key, None)
```

I'm really not sure that is any better.

Reply:
My suggestion to the frontend would remove the key as it does when it's not set from the beginning (which is my expectation and not set it to pass `None` or anything else).
Regardless it needs to be handled and would personally think best option is to do it in the config flow to pass a correct entry data